### PR TITLE
Add more context to completion

### PR DESCRIPTION
**Pull request by bthesorceror**
- Trigger type from parsed flux
- Bucket for current function

### CODE REVIEW

File: src/handlers/completion.rs [hunk 1]
```diff
@@ -147,30 +192,90 @@ async fn get_user_matches(
     Ok(result)
 }
 
+fn follow_pipes_for_bucket(call: Box<CallExpr>) -> Option<String> {
+    for arg in call.arguments {
+        if arg.key.name == "bucket" {
```
Comment (jsteenb2) [inline src/handlers/completion.rs:197]: could this have false positives if someone creates a function with a bucket input param as an arg?

File: src/handlers/completion.rs [hunk 2]
```diff
@@ -147,30 +192,90 @@ async fn get_user_matches(
...
-        let mut user_matches =
-            get_user_matches(uri, pos, ctx).await?;
+                let mut user_matches =
+                    get_user_matches(uri, pos, ctx).await?;
 
-        items.append(&mut user_matches);
+                items.append(&mut user_matches);
+            }
+            CompletionType::Logical => {
+                if info.ident == "measurement" {
```
Comment (jsteenb2) [inline src/handlers/completion.rs:271]: nice, like the use of completion types here. Easy to follow what this is doing

File: src/macros/mod.rs [hunk 3]
```diff
@@ -0,0 +1,6 @@
+#[macro_export]
+macro_rules! log {
+    ( $( $t:tt )* ) => {
+        web_sys::console::log_1(&format!( $( $t )* ).into());
```
Comment (jsteenb2) [inline src/macros/mod.rs:4]: does this `console.log` in the wasm target?

Comment (bthesorceror) [reply src/macros/mod.rs:4]: yeah, added it for debugging purposes

File: src/handlers/completion.rs (hunk 1 above)
Comment (bthesorceror) [reply src/handlers/completion.rs:197]: absolutely, but we don't have a better solution at the moment.

**Pull request merged**